lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info = {})
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if self.services.include?("Vehicle Registration") && vehicle.antique? == true
+      @registered_vehicles << vehicle
+      @collected_fees += 25
+      vehicle.plate_type = :antique
+      @registered_vehicles
+    elsif self.services.include?("Vehicle Registration") && vehicle.electric_vehicle? == true
+      @registered_vehicles << vehicle
+      @collected_fees += 200
+      vehicle.plate_type = :ev
+      @registered_vehicles
+    elsif self.services.include?("Vehicle Registration")
+      @registered_vehicles << vehicle
+      @collected_fees += 100
+      vehicle.plate_type = :regular
+      @registered_vehicles
+    else
+      nil
+    end
+  end
+
+  def administer_written_test(registrant)
+    if registrant.permit? == true && registrant.age >= 16 && self.services.include?("Written Test")
+      registrant.license_data[:written] = true
+      true
```

Comment:
I think you can get ride of line 41. Also you don't need to call `self.services` use the instance variable `@services.include?("Written Test")`